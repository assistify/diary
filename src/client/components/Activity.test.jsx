import React from 'react';
import renderer from 'react-test-renderer';
import ActivityItems from './Activity';

it('renders a list correctly', () => {
  const tree = renderer
    .create(
      <ActivityItems
        title="sample"
        list={[
          { title: 'sample title 1', details: 'sample details' },
          { title: 'sample title 2', details: 'sample details' },
        ]}
        className="sample"
        contentEditable={false}
        updateValue={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('does not render an empty list', () => {
  const tree = renderer
    .create(
      <ActivityItems
        title="empty"
        list={[]}
        className="sample"
        contentEditable={false}
        updateValue={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('does not render a list without titled items', () => {
  const tree = renderer
    .create(
      <ActivityItems
        title="no-title"
        list={[{ title: '' }]}
        className="sample"
        contentEditable={false}
        updateValue={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
