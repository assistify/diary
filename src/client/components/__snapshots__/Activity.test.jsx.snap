// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render a list without titled items 1`] = `null`;

exports[`does not render an empty list 1`] = `null`;

exports[`renders a list correctly 1`] = `
<div
  className="box c-activities sample"
  style={Object {}}
>
  <h1
    className="title is-6"
  >
    sample
  </h1>
  <ul>
    <li>
      <div
        contentEditable={false}
        dangerouslySetInnerHTML={
          Object {
            "__html": "sample title 1",
          }
        }
        onBlur={[Function]}
      />
    </li>
    <li>
      <div
        contentEditable={false}
        dangerouslySetInnerHTML={
          Object {
            "__html": "sample title 2",
          }
        }
        onBlur={[Function]}
      />
    </li>
  </ul>
</div>
`;
